robot_skills/src/robot_skills/robot_part.py

```diff
@@ -190,3 +193,10 @@ def process_hardware_status(self, diagnostic_dict):
             else:
                 self._operational = True
 
+    @abc.abstractmethod
```

Comment:
Oooh, fancy, metaclasses!
But if there is an reset-method that is inherited by all RobotPart-subclasses, why implement it everywhere? I'd say implement it only when the default ```pass``` implementation should be overridden.

Reply:
So you're saying "Don't define `reset` as an abstract method, but implement it in the base class as `pass`, and override if necessary in the child classes"?

Reply:
Sounds good! We implemented the suggestion to '''pass''' in the default function, and only override this where necessary, on Hero this works, but on Amigo it doesn't so we'll try to check where this goes wrong.